sshagent_windows.go

```diff
@@ -50,29 +57,46 @@ func Available() bool {
 }
 
 // New returns a new agent.Agent and the (custom) connection it uses
-// to communicate with a running pagent.exe instance (see README.md)
+// to communicate with a running pagent.exe instance (see README.md).
 func New() (agent.Agent, net.Conn, error) {
 	if pageantWindow() != 0 {
-		return agent.NewClient(&conn{}), nil, nil
+		return agent.NewClient(&conn{}), &conn{}, nil
+	}
+
+	sshAgentPipe := openSSHAgentPipe
+	if sshAuthSock := os.Getenv("SSH_AUTH_SOCK"); sshAuthSock != "" {
+		conn, err := net.Dial("unix", sshAuthSock)
+		if err == nil {
+			return agent.NewClient(conn), conn, nil
+		}
+
+		if !strings.HasPrefix(sshAuthSock, pipe) {
+			sshAuthSock = pipe + sshAuthSock
+		}
+
+		sshAgentPipe = sshAuthSock
 	}
+
 	conn, err := winio.DialPipe(sshAgentPipe, nil)
 	if err != nil {
 		return nil, nil, errors.New(
 			"SSH agent requested, but could not detect Pageant or Windows native SSH agent",
 		)
 	}
-	return agent.NewClient(conn), nil, nil
+
+	return agent.NewClient(conn), conn, nil
 }
 
 type conn struct {
 	sync.Mutex
 	buf []byte
 }
 
-func (c *conn) Close() {
+func (c *conn) Close() error {
```

Comment:
I assume this is also done to mimic `net.Conn`? Inline with my other comment I suggest we do not mask this `conn` type so we can return it as some kind a fake `net.Conn`.